app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb

```diff
@@ -77,76 +77,135 @@ def perf_collect_metrics(interval_name, start_time = nil, end_time = nil)
       return
     end
 
-    end_time   = end_time   ? end_time.utc   : Time.now.utc
-    start_time = start_time ? start_time.utc : (end_time - 4.hours) # 4 hours for symmetry with VIM
+    end_time       = end_time   ? end_time.to_time.utc   : Time.now.utc
+    start_time     = start_time ? start_time.to_time.utc : (end_time - 4.hours) # 4 hours for symmetry with VIM
+    time_interval  = start_time..end_time
+    counter_values = {}
 
     # This is just for consistency, to produce a :connect benchmark
     Benchmark.realtime_block(:connect) {}
 
     ems.with_provider_connection do |connection|
-      metrics_conn = Azure::Armrest::Insights::MetricsService.new(connection)
+      # TODO: code smell - should be in a 'collector' class instantiated here / find a better solution?
+      @connection = connection
+      @start_time = start_time.iso8601
+      @end_time   = end_time.iso8601
```

Comment:
Similar comment here, this already is an instance of the MetricsCapture class so idk why you'd need another collector class, but also having method args be instance vars to avoid passing arguments to another method feels wrong.

Reply:
because current class doesn't has this variables on initialization, they come here through `perf_collect_metrics` method, not via `initialize`.

Reply:
done